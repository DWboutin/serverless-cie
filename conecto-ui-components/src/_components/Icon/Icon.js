import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import To from '../Logo/Logos/To'
import Menu from './Icons/Menu'
import Alert from './Icons/Alert'
import ArrowDown from './Icons/ArrowDown'
import Bookmark from './Icons/Bookmark'
import Check from './Icons/Check'
import Close from './Icons/Close'
import Cog from './Icons/Cog'
import Concurrents from './Icons/Concurrents'
import Email from './Icons/Email'
import Facebook from './Icons/Facebook'
import Flag from './Icons/Flag'
import ColoredFlag from './Icons/ColoredFlag'
import Marker from './Icons/Marker'
import Notification from './Icons/Notification'
import Phone from './Icons/Phone'
import Search from './Icons/Search'
import SMS from './Icons/SMS'
import ThumbUp from './Icons/ThumbUp'
import User from './Icons/User'

import CCAmex from './Icons/CCAmex'
import CCDefault from './Icons/CCDefault'
import CCDinersClub from './Icons/CCDinersClub'
import CCDiscover from './Icons/CCDiscover'
import CCJCB from './Icons/CCJCB'
import CCMastercard from './Icons/CCMastercard'
import CCVisa from './Icons/CCVisa'

class Icon extends React.Component {
  constructor(props) {
    super(props)

    this.classesPrefix = 'to-icon'
  }

  iconDefinition(type) {
    switch (type) {
      case 'alert':
        return Alert
      case 'arrow-down':
        return ArrowDown
      case 'bookmark':
        return Bookmark
      case 'check':
        return Check
      case 'close':
        return Close
      case 'cog':
        return Cog
      case 'concurrents':
        return Concurrents
      case 'email':
        return Email
      case 'facebook':
        return Facebook
      case 'flag':
        return Flag
      case 'colored-flag':
        return ColoredFlag
      case 'marker':
        return Marker
      case 'menu':
        return Menu
      case 'notification':
        return Notification
      case 'phone':
        return Phone
      case 'search':
        return Search
      case 'sms':
        return SMS
      case 'thumb-up':
        return ThumbUp
      case 'user':
        return User
      case 'cc-amex':
        return CCAmex
      case 'cc-default':
        return CCDefault
      case 'cc-dinersclub':
        return CCDinersClub
      case 'cc-discover':
        return CCDiscover
      case 'cc-jcb':
        return CCJCB
      case 'cc-mastercard':
        return CCMastercard
      case 'cc-visa':
        return CCVisa
      case 'conecto':
      default:
        return To
    }
  }

  render() {
    const { className, type } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      [`${className}`]: className,
    })
    const iconToRender = this.iconDefinition(type)

    return (
      <i className={classes} aria-label={`icon: ${type}`}>
        {React.createElement(iconToRender)}
      </i>
    )
  }
}

Icon.defaultProps = {
  type: 'conecto',
}

const IconTypePropTypes = PropTypes.oneOf([
  'alert',
  'arrow-down',
  'bookmark',
  'check',
  'close',
  'cog',
  'concurrents',
  'email',
  'facebook',
  'flag',
  'colored-flag',
  'marker',
  'menu',
  'notification',
  'phone',
  'search',
  'sms',
  'thumb-up',
  'user',
  'cc-amex',
  'cc-default',
  'cc-dinersclub',
  'cc-discover',
  'cc-jcb',
  'cc-mastercard',
  'cc-visa',
  'conecto',
])

Icon.propTypes = {
  type: IconTypePropTypes,
  className: PropTypes.string,
}

export { Icon as default, IconTypePropTypes }
