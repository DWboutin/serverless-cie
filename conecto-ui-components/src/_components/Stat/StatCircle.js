import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { useCountUp } from 'react-countup'

import { CardStat } from '../Card'

const StatCircle = ({ className, icon, title, notice, number }) => {
  const { countUp } = useCountUp({
    start: 0,
    end: number,
    delay: 0,
    duration: 2,
  })
  const classes = classNames({
    'to-stat-circle': true,
    [`${className}`]: className,
  })

  return (
    <CardStat className={classes} icon={icon} title={title}>
      <div
        className={`progress-circle p${countUp} ${countUp >= 50 && 'over50'}`}
      >
        <span>{countUp}%</span>
        <div className="left-half-clipper">
          <div className="first50-bar" />
          <div className="value-bar" />
        </div>
      </div>
      {notice && <div className="notice">{notice}</div>}
    </CardStat>
  )
}

StatCircle.defaultProps = {}

StatCircle.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  notice: PropTypes.string,
}

export default StatCircle
