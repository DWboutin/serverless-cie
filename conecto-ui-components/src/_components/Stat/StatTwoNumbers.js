import React from 'react'
import PropTypes from 'prop-types'
import omit from 'omit.js'
import classNames from 'classnames'
import CountUp from 'react-countup'

import { CardStat } from '../Card'

const StatTwoNumbers = ({
  className,
  icon,
  title,
  number1,
  number2,
  notice1,
  notice2,
  suffix1,
  suffix2,
  disabled1,
  disabled2,
}) => {
  const classes = classNames({
    'to-stat-two-numbers': true,
    [`${className}`]: className,
  })

  return (
    <CardStat className={classes} icon={icon} title={title}>
      <div className={`number ${disabled1 && 'disabled'}`}>
        <div className="card">
          <span>
            <CountUp end={number1} duration="2" />
            {suffix1}
          </span>
        </div>
        <div className="notice">{notice1}</div>
      </div>
      <div className={`number ${disabled2 && 'disabled'}`}>
        <div className="card">
          <span>
            <CountUp end={number2} duration="2" />
            {suffix2}
          </span>
        </div>
        <div className="notice">{notice2}</div>
      </div>
    </CardStat>
  )
}

StatTwoNumbers.defaultProps = {
  disabled1: false,
  disabled2: false,
  suffix1: '',
  suffix2: '',
  notice1: '',
  notice2: '',
}

StatTwoNumbers.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  number1: PropTypes.number.isRequired,
  suffix1: PropTypes.string.isRequired,
  notice1: PropTypes.string.isRequired,
  disabled1: PropTypes.bool.isRequired,
  number2: PropTypes.number.isRequired,
  suffix2: PropTypes.string.isRequired,
  notice2: PropTypes.string.isRequired,
  disabled2: PropTypes.bool.isRequired,
}

export default StatTwoNumbers
