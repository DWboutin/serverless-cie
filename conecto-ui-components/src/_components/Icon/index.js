import Icon, { IconTypePropTypes } from './Icon'

export { Icon as default, IconTypePropTypes }
