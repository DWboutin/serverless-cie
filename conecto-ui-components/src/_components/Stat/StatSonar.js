import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import CountUp from 'react-countup'

import { CardStat } from '../Card'
import Icon from '../Icon'

const StatSonar = ({ className, icon, title, number, suffix }) => {
  const classes = classNames({
    'to-stat-sonar': true,
    [`${className}`]: className,
  })

  return (
    <CardStat className={classes} icon={icon} title={title}>
      <div className="sonar">
        <div className="inner-sonar">
          <div className="number">
            <span>
              <CountUp end={number} duration="2" />
              {suffix}
            </span>
          </div>
        </div>
        <div className="icon-sonar">
          <div className="icon-sonar-wrap">
            <span>
              <Icon type="marker" />
            </span>
          </div>
        </div>
      </div>
    </CardStat>
  )
}

StatSonar.defaultProps = {}

StatSonar.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  suffix: PropTypes.string,
}

export default StatSonar
