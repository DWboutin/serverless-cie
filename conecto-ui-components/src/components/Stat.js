import React from 'react'

import Grid, { Row, Col } from '../_components/Grid'
import Icon from '../_components/Icon'
import StatCircle, { StatSonar, StatTwoNumbers } from '../_components/Stat'

const Stat = () => (
  <Grid>
    <Row>
      <Col>
        <h2 style={{ opacity: 0.7 }}>Stat</h2>
      </Col>
    </Row>
    <Row>
      <Col
        xs={{
          span: 12,
        }}
        sm={{
          span: 12,
        }}
        md={{
          span: 4,
        }}
        lg={{
          span: 4,
        }}
        xl={{
          span: 4,
        }}
      >
        <h3 style={{ opacity: 0.6, marginBottom: '40px' }}>Stat Circle</h3>
        <StatCircle
          icon={<Icon type="notification" />}
          title="Taux d'accceptation"
          notice="+ 6.5% / mois précédent"
          number={66}
        />
      </Col>
      <Col
        xs={{
          span: 12,
        }}
        sm={{
          span: 12,
        }}
        md={{
          span: 4,
        }}
        lg={{
          span: 4,
        }}
        xl={{
          span: 4,
        }}
      >
        <h3 style={{ opacity: 0.6, marginBottom: '40px' }}>Stat Two Numbers</h3>
        <StatTwoNumbers
          icon={<Icon type="notification" />}
          title="Taux d'accceptation"
          notice="+ 6.5% / mois précédent"
          number1={45}
          suffix1="%"
          disabled1
          notice1="Courriel"
          number2={66}
          suffix2="h"
          notice2="Téléphone"
        />
      </Col>
      <Col
        xs={{
          span: 12,
        }}
        sm={{
          span: 12,
        }}
        md={{
          span: 4,
        }}
        lg={{
          span: 4,
        }}
        xl={{
          span: 4,
        }}
      >
        <h3 style={{ opacity: 0.6, marginBottom: '40px' }}>Stat Sonar</h3>
        <StatSonar
          icon={<Icon type="notification" />}
          title="Taux d'accceptation"
          notice="+ 6.5% / mois précédent"
          number={60}
          suffix="h"
        />
      </Col>
    </Row>
  </Grid>
)

Stat.defaultProps = {}

export default Stat
