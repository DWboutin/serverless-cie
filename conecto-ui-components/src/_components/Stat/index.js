import StatCircle from './StatCircle'
import StatTwoNumbers from './StatTwoNumbers'
import StatSonar from './StatSonar'

export { StatCircle as default, StatTwoNumbers, StatSonar }
