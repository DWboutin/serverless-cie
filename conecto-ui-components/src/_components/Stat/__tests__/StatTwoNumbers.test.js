import React from 'react'
import { shallow, mount } from 'enzyme'

import { StatTwoNumbers } from '../../Stat'
import Icon from '../../Icon'

describe('StatTwoNumbers', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <StatTwoNumbers
        icon={<Icon type="notification" />}
        title="Taux d'accceptation"
        notice="+ 6.5% / mois précédent"
        number1={45}
        suffix1="%"
        disabled1
        notice1="Courriel"
        number2={66}
        suffix2="h"
        notice2="Téléphone"
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should have the prefix class', () => {
    const wrapper = shallow(
      <StatTwoNumbers
        icon={<Icon type="notification" />}
        title="Taux d'accceptation"
        notice="+ 6.5% / mois précédent"
        number1={45}
        suffix1="%"
        disabled1
        notice1="Courriel"
        number2={66}
        suffix2="h"
        notice2="Téléphone"
      />
    )

    expect(wrapper.hasClass('to-stat-two-numbers')).toEqual(true)
  })
})
