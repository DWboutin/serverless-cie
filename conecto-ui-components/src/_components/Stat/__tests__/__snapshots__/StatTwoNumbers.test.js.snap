// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatTwoNumbers should match snapshot 1`] = `
<CardStat
  className="to-stat-two-numbers"
  icon={
    <Icon
      type="notification"
    />
  }
  title="Taux d'accceptation"
  type="bordered"
>
  <div
    className="number disabled"
  >
    <div
      className="card"
    >
      <span>
        <CountUp$$1
          decimal="."
          decimals={0}
          delay={null}
          duration="2"
          easingFn={null}
          end={45}
          formattingFn={null}
          onEnd={[Function]}
          onPauseResume={[Function]}
          onReset={[Function]}
          onStart={[Function]}
          onUpdate={[Function]}
          prefix=""
          redraw={false}
          separator=""
          start={0}
          suffix=""
          useEasing={true}
        />
        %
      </span>
    </div>
    <div
      className="notice"
    >
      Courriel
    </div>
  </div>
  <div
    className="number false"
  >
    <div
      className="card"
    >
      <span>
        <CountUp$$1
          decimal="."
          decimals={0}
          delay={null}
          duration="2"
          easingFn={null}
          end={66}
          formattingFn={null}
          onEnd={[Function]}
          onPauseResume={[Function]}
          onReset={[Function]}
          onStart={[Function]}
          onUpdate={[Function]}
          prefix=""
          redraw={false}
          separator=""
          start={0}
          suffix=""
          useEasing={true}
        />
        h
      </span>
    </div>
    <div
      className="notice"
    >
      Téléphone
    </div>
  </div>
</CardStat>
`;
